source/MenuPanel.cpp

```diff
@@ -54,7 +54,7 @@ namespace {
 
 
 
-MenuPanel::MenuPanel(PlayerInfo &player, UI &gamePanels)
+MenuPanel::MenuPanel(PlayerInfo &player, UI &gamePanels, bool pauseAudio)
```

Comment:
Pausing audio doesn't feel like a property of a Panel, it seems more like a property of a stack of panels (if there is anything on the stack of panels, then pause audio), or a property of the engine being active or not. (pause audio if the engine is not active.)

Reply:
Unfortunately, there isn't an easy way of doing it.

Engine::Step has an `active` parameter, but that's only false if a pop-up panel pauses the engine... if we are in the menu (by pressing Escape, for instance), the entire Engine::Step call is skipped.

main.cpp has an `isPaused` property, but that only exists there, and isn't passed to the engine. The engine's `active` parameter in Engine::Step is stored in MainPanel, which we can't easily access from main.cpp. Audio updates are only done in main and Engine.

I don't really see a way of managing both of these options.

Reply:
Isn't it possible to just always pause the audio when the MenuPanel gets opened? (So always performing Audio::Pause() in the constructor of MenuPanel, without the need for this variable in the constructor?)

Reply:
Almost. The only case we need this special handling is when the panel opens up the first time, as the initial animation panel plays a sound that we shouldn't pause.